### ISSUE TITLE
Destructured immediately-invoked function with spread raises false errors

### ISSUE THREAD

**Issue by laverdet** [microsoft/TypeScript#49585]
# Bug Report

### 🔎 Search Terms
destructured spread iife 

### 🕗 Version & Regression Information
- This changed between versions v4.4.4 and v4.5.5

### ⏯ Playground Link
[Playground link with relevant code](https://www.typescriptlang.org/play?ts=4.7.2#code/MYewdgzgLgBA3jAbgQwDYFcCmMC+MC8MAZumMFAJbgAUAlPALABQMrMATplOu2Iy20EoMmAFwwARBIA0zQYIB0SqOywwA-PDzi4OWQNY4A3Mxx0TTIA)

### 💻 Code
```ts
const { value } = function() {
    return {
        value: "",
        ...true ? {} : {},
    };
}();
```

### 🙁 Actual behavior
This raises two errors:
- Line 1: Initializer provides no value for this binding element and the binding element has no default value.
- Line 3: 'value' is specified more than once, so this usage will be overwritten.

If you remove the `...true ? {} : {}` spread then it compiles successfully.

### 🙂 Expected behavior
I wouldn't expect any errors in this code. Especially the second error which is definitely not true.
